StockSyncAdapter/app/src/main/java/shuvalov/nikita/stocksyncadapter/StockSyncAdapter.java

```diff
@@ -0,0 +1,132 @@
+
+import android.accounts.Account;
+import android.content.AbstractThreadedSyncAdapter;
+import android.content.ContentProviderClient;
+import android.content.ContentResolver;
+import android.content.ContentValues;
+import android.content.Context;
+import android.content.SyncResult;
+import android.database.Cursor;
+import android.os.Bundle;
+import android.os.RemoteException;
+import android.util.Log;
+
+import com.android.volley.Request;
+import com.android.volley.RequestQueue;
+import com.android.volley.Response;
+import com.android.volley.VolleyError;
+import com.android.volley.toolbox.JsonArrayRequest;
+import com.android.volley.toolbox.JsonObjectRequest;
+import com.android.volley.toolbox.Volley;
+
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import java.util.ArrayList;
+
+
+/**
+ * Created by NikitaShuvalov on 12/8/16.
+ */
+
+public class StockSyncAdapter extends AbstractThreadedSyncAdapter {
+    private Context mContext;
+
+    public StockSyncAdapter(Context context, boolean autoInitialize) {
+        super(context, autoInitialize);
+        mContext = context;
+    }
+
+    public StockSyncAdapter(Context context, boolean autoInitialize, boolean allowParallelSyncs) {
+        super(context, autoInitialize, allowParallelSyncs);
+        mContext = context;
+    }
+
+    @Override
+    public void onPerformSync(Account account, Bundle bundle, String s, ContentProviderClient contentProviderClient, SyncResult syncResult) {
+        Cursor cursor=null;
+        Log.d("OnSyncPerform", "s");
+        try {
+            cursor = contentProviderClient.query(StockPortfolioContract.Stocks.CONTENT_URI,
+                    null,
+                    null,
+                    null,
+                    null,
+                    null);
+        } catch (RemoteException e) {
+            e.printStackTrace();
+        }
+        ArrayList<String> stockSymbols = new ArrayList<>();
+        if(cursor.moveToFirst()){
+            while(!cursor.isAfterLast()){
+                stockSymbols.add(cursor.getString(cursor.getColumnIndex(StockPortfolioContract.Stocks.COLUMN_STOCK_SYMBOL)));
+                cursor.moveToNext();
+            }
+        }
+
+        for(String symbol: stockSymbols) {
+            RequestQueue queue = Volley.newRequestQueue(mContext);
+            String quoteUrl = "http://dev.markitondemand.com/MODApis/Api/v2/Quote/json?symbol=" + symbol;
+
+            final JsonObjectRequest quoteJsonRequest = new JsonObjectRequest
+                    (Request.Method.GET, quoteUrl, null, new Response.Listener<JSONObject>() {
+                        @Override
+                        public void onResponse(JSONObject response) {
+                            Log.d(MainActivity.class.getName(), "Response2: " + response.toString());
+                            try {
+                                ContentResolver contentResolver = mContext.getContentResolver();
```

Comment:
Don't save a reference to a `Context` as a member variable if you can avoid it. Objects that extend `Context` (e.g. activities or services) are large and we don't want to prevent them from being garbage collected by saving references to them that we don't need. In this case, just use the context to create the `ContentResolver` in the constructor and save _it_ as a member variable instead.